.vib/harbor-jobservice/goss/harbor-jobservice.yaml

```diff
@@ -0,0 +1,17 @@
+group:
+  daemon:
+    exists: true
+user:
+  daemon:
+    exists: true
+command:
+  check-system-cert-paths:
+    exec: if (test -d /etc/ssl/certs/ && (ls -ld /etc/ssl/certs/ | cut --delimiter=' ' --fields=3 | grep 'harbor') && (ls -ld /etc/ssl/certs/ | cut --delimiter=' ' --fields=1 | grep 'drwxrwxr-x')); then echo "Directory exists"; fi || if (test -d /etc/pki/tls/certs/ && (ls -ld /etc/pki/tls/certs/ | cut --delimiter=' ' --fields=3 | grep 'harbor') && (ls -ld /etc/pki/tls/certs/ | cut --delimiter=' ' --fields=1 | grep 'drwxrwxr-x')); then echo "Directory exists"; fi
+    exit-status: 0
+    stdout:
+      - "Directory exists"
+  check-permissions-system-certs:
+    exec: if (test -f /etc/ssl/certs/ca-certificates.crt && (ls -ld /etc/ssl/certs/ca-certificates.crt | cut --delimiter=' ' --fields=3 | grep 'harbor')); then echo "File exists"; fi || if (test -f /etc/pki/tls/certs/ca-bundle.crt && (ls -ld /etc/pki/tls/certs/ca-bundle.crt | cut --delimiter=' ' --fields=3 | grep 'harbor')); then echo "File exists"; fi || if (test -f /etc/pki/tls/certs/ca-bundle.trust.crt && (ls -ld /etc/pki/tls/certs/ca-bundle.trust.crt | cut --delimiter=' ' --fields=3 | grep 'harbor')); then echo "File exists"; fi
+    exit-status: 0
+    stdout:
+      - "File exists"
```

Comment:
I would also modify this. This would be a bit more tricky, but modifying the command from above and making use of `stat` should suffice. As it is, both execs are very confusing.

Additionally, you should add a comment briefly explaining why these tests are needed to avoid guesswork in the future

Reply:
The idea is not only check that the files exists but also check the user of the files if they exist:
```
configure_permissions_system_certs() {
    local -r owner="${1:-}"
    # Debian
    set_permissions_ownership "/etc/pki/tls/certs/ca-bundle.crt" "$owner"
    # Centos/Phonton
    set_permissions_ownership "/etc/pki/tls/certs/ca-bundle.trust.crt" "$owner"
    set_permissions_ownership "/etc/ssl/certs/ca-certificates.crt" "$owner"
}
```
Code from libharbor.sh file.

Reply:
This is done to check the changes performed by running `configure_permissions_system_certs` right? Unless I'm mistaken, that code is not actually modifying the file's users:

```
...
configure_permissions_system_certs() {
    local -r owner="${1:-}"
    # Debian
    set_permissions_ownership "/etc/pki/tls/certs/ca-bundle.crt" "$owner"
    # Centos/Phonton
    set_permissions_ownership "/etc/pki/tls/certs/ca-bundle.trust.crt" "$owner"
    set_permissions_ownership "/etc/ssl/certs/ca-certificates.crt" "$owner"
}
...
set_permissions_ownership() {
    local -r path="${1:?path is missing}"
    local -r owner="${2:-}"

    if [[ -f "$path" ]]; then
        chmod g+w "$path"
        if [[ -n "$owner" ]]; then
            chown "$owner" "$path"
        fi
    fi
}
...
```

We are not specifying an owner when calling `configure_permissions_system_certs`, so the only change done there is a permissions change. In that case, we would need to look at the permissions instead.